src/core.ts

```diff
@@ -1,36 +1,36 @@
+import * as u from './util'
 var Muxrpc = require('muxrpc')
 var pull = require('pull-stream')
 // var Rate = require('pull-rate')
-var u = require('./util')
 var MultiServer = require('multiserver')
 var Inactive = require('pull-inactivity')
 var debug = require('debug')('secret-stack')
 
-function isFunction (f) {
+function isFunction (f: any): f is Function {
   return typeof f === 'function'
 }
 
-function isString (s) {
+function isString (s: any): s is string {
   return typeof s === 'string'
 }
 
-function isObject (o) {
+function isObject (o: any): any {
   return o && typeof o === 'object' && !Array.isArray(o)
 }
 
 var isArray = Array.isArray
 
-function toBase64 (s) {
+function toBase64 (s: Buffer | string) {
   if (isString(s)) return s
-  else s.toString('base64') // assume a buffer
+  else return s.toString('base64') // assume a buffer
```

Comment:
Here's an example of bug that TS caught. This function was returning undefined, when it should return a string always